vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/main/java/com/vaadin/flow/component/treegrid/it/TreeGridStringDataGenerator.java

```diff
@@ -0,0 +1,62 @@
+package com.vaadin.flow.component.treegrid.it;
+
+import com.vaadin.flow.data.provider.hierarchy.TreeData;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class TreeGridStringDataGenerator {
+
+    private final TreeData<String> data = new TreeData<>();
+
+    private final Map<String, String> parentPathMap = new HashMap<>();
+
+    private Map<Integer, String> levelToNameMap;
+
+    public TreeGridStringDataGenerator() {
+        this(Map.of(0, "Granddad", 1, "Dad", 2, "Son"));
+    }
+
+    public TreeGridStringDataGenerator(Map<Integer, String> levelToNameMap) {
```

Comment:
Could we use an array instead of a Map here?

```java
new TreeGridStringDataGenerator("Graddad", "Dad", "Son").generate(3, 3, 3)
```

Reply:
Yes, using a map is definitely unnecessary. I will change that. For names, if they are required in the constructor and don't have a default, I think then builder approach would make more sense. I don't have a strong opinion on which option to choose. It can also be modified somewhat easily afterwards. If you think that enforcing setting the names explicitly would make it clearer and more future-proof, I will convert it to builder.

Reply:
Well, now I'm thinking that the Builder approach might be a better option after all. I checked and the item names were explicit before but now we are hiding them.

Reply:
I updated the branch

Reply:
Thanks, looks good now. 👍